hyperspy/_signals/eels.py

```diff
@@ -224,8 +241,11 @@ def align_zero_loss_peak(
         def substract_from_offset(value, signals):
             for signal in signals:
                 signal.axes_manager[-1].offset -= value
-
-        zlpc = self.estimate_zero_loss_peak_centre(mask=mask)
+        
+        if signal_range:
+            zlpc = self[...,signal_range[0]:signal_range[1]].estimate_zero_loss_peak_centre(mask=mask)
```

Comment:
I propose the following equivalent but more readable syntax (see #388 for a discussion):

```python

zlpc = self.isig[signal_range[0]:signal_range[1]].estimate_zero_loss_peak_centre(mask=mask)

```